mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageResultIterator.java

```diff
@@ -253,7 +247,7 @@ public boolean equals(Object obj) {
             }
             if (obj instanceof UnloadedMessageResult) {
                 UnloadedMessageResult that = (UnloadedMessageResult)obj;
-                return (size == that.size) && (uid == that.uid) && (modSeq == that.modSeq) && exception.equals(that.exception)
+                return (size == that.size) && (uid.equals(that.uid)) && (modSeq == that.modSeq) && exception.equals(that.exception)

```

Comment:
It could, it's not what I decided to do, I preferred not refactor that piece of code.